eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/AbstractMutableList.java

```diff
@@ -1116,6 +1117,29 @@ public int binarySearch(T key)
         return Collections.binarySearch((List<? extends Comparable<? super T>>) this, key);
     }
 
+    public MutableList<RichIterable<T>> chunk(int size)
+    {
+        if (size <= 0)
+        {
+            throw new IllegalArgumentException("Size for groups must be positive but was: " + size);
+        }
+
+        MutableList<RichIterable<T>> result = Lists.mutable.empty();
+        int i = 0;
+        while (i < this.size())
+        {
+            MutableList<T> batch = new FastList<T>(Math.min(size, this.size() - i));
+
+            for (int j = 0; j < size && i < this.size(); j++)
+            {
+                batch.add(this.get(i));
```

Comment:
This assumes that all subclasses of AbstractMutableList implement RandomAccess. I think we're inconsistent about this assumption today and ought to split AbstractMutableList into two classes.